### ISSUE TITLE
Issue using adapt framework with tensorflow GPU!

### ISSUE THREAD

**Issue by sreenivasaupadhyaya** [adapt-python/adapt#61]
I am trying to install adapt in a framework with tensorflow and system with nvidia GPUs.

I get this error after installing adapt framework when importing tensorflow. 

"import tensorflow
2022-07-08 15:46:48.535030: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory
2022-07-08 15:46:48.535055: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine."

On looking up in the internet, itsdue to the system looks for libcudart.so.11.0 when its not present. One work around suggested is to do linking of another version of cuda library other than "libcudart.so.11.0". My question is that why this happens after the installation of adapt.
I tried to install adapt with different version of tensorflow but the same issue exists.
Any guesses?

**Comment by antoinedemathelin** [comment]
Hi @sreenivasaupadhyaya,

I know this issue but I don't think it comes from adapt but from Tensorflow.
Which Python version and Tensorflow version have you tried?

I already managed to use adapt with Tensorflow on GPUs with Python 3.6 and Tensorflow 2.3.1, with Python 3.8 I never succeed to use GPUs with Tensorflow.

Best,

**Comment by sreenivasaupadhyaya** [comment]
python version is 3.6.8 and tensorflow i tried with 2.0.0, 2.3.0 and 2.4.0

**Comment by antoinedemathelin** [comment]
Hi @sreenivasaupadhyaya,
Strange indeed, when you install Tensorflow alone whithout adapt you don't get this warning? And then when you install adapt you get it?

Did you try this to see if Tensorflow find your GPUs?

```
from tensorflow.python.client import device_lib

device_lib.list_local_devices()
```

**Comment by sreenivasaupadhyaya** [comment]
Hi @antoinedemathelin 

You are correct.
Please find the pip freeze files before and after i install the adapt package.

Afterinstalling adapt  and running the commands as you suggested i get this

`Python 3.6.9 (default, Apr 18 2020, 01:56:04) 
[GCC 8.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from tensorflow.python.client import device_lib
2022-07-11 19:32:32.117033: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory
2022-07-11 19:32:32.117056: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
>>> device_lib.list_local_devices()
2022-07-11 19:32:35.187880: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-07-11 19:32:35.550271: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory
2022-07-11 19:32:35.550405: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcublas.so.11'; dlerror: libcublas.so.11: cannot open shared object file: No such file or directory
2022-07-11 19:32:35.550475: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcublasLt.so.11'; dlerror: libcublasLt.so.11: cannot open shared object file: No such file or directory
2022-07-11 19:32:35.552255: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcusolver.so.11'; dlerror: libcusolver.so.11: cannot open shared object file: No such file or directory
2022-07-11 19:32:35.552323: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcusparse.so.11'; dlerror: libcusparse.so.11: cannot open shared object file: No such file or directory
2022-07-11 19:32:35.552387: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudnn.so.8'; dlerror: libcudnn.so.8: cannot open shared object file: No such file or directory
2022-07-11 19:32:35.552398: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1835] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
[name: "/device:CPU:0"
device_type: "CPU"
memory_limit: 268435456
locality {
}
incarnation: 9497802731932743444
]
>>> `

**Comment by sreenivasaupadhyaya** [comment]
@antoinedemathelin ,

I just tried with tensorflow 2.3.1 and its showing the gpus. I will keep you updated if i am able to use the adapt framework.
Keep you posted!

**Comment by antoinedemathelin** [comment]
Hi @sreenivasaupadhyaya,

Ok, from what I see in the file you send me, tensorflow is not already installed before you install adapt, so when you install adapt, it looks for the latest version of tensorflow available for Python 3.6 and install it. Thus you get tensorflow 2.6.2 which does not find the GPUs. By downgrading tensorflow to 2.3.1 it should be fine indeed.

**Comment by sreenivasaupadhyaya** [comment]
> Hi @sreenivasaupadhyaya,
> 
> Ok, from what I see in the file you send me, tensorflow is not already installed before you install adapt, so when you install adapt, it looks for the latest version of tensorflow available for Python 3.6 and install it. Thus you get tensorflow 2.6.2 which does not find the GPUs. By downgrading tensorflow to 2.3.1 it should be fine indeed.

Thanks @antoinedemathelin , I was able to use the GPU.  I am closing this ticket.
